src/send-and-sync.md

```diff
@@ -74,7 +74,171 @@ of their pervasive use of raw pointers to manage allocations and complex ownersh
 Similarly, most iterators into these collections are Send and Sync because they
 largely behave like an `&` or `&mut` into the collection.
 
+## Example
+
+[`Box`][box-doc] is implemented as it's own special intrinsic type by the
+compiler for [various reasons][box-is-special], but we can implement something
+with similar-ish behavior ourselves to see an example of when it is sound to
+implement Send and Sync. Let's call it a `Carton`.
+
+We start by writing code to take a value allocated on the stack and transfer it
+to the heap.
+
+```rust,ignore
+use std::mem::size_of;
+use std::ptr;
+
```

Comment:
Let's try to have this code be `cargo check`ed by mocking `libc`:
```suggestion
```rust
# mod libc {
#     pub use ::std::os::raw::{c_int, c_void, size_t};
#     extern "C" { pub fn posix_memalign(memptr: *mut *mut c_void, align: size_t, size: size_t) -> c_int; }
# }
use std::{
    mem::{align_of, size_of},
    ptr,
};
use libc::c_void;

```